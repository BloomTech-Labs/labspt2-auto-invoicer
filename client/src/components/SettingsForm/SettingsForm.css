.settings_form{
    display: flex;
    max-width: 600px;
    width:100%;
    align-items: center;
    height:400px;
}

