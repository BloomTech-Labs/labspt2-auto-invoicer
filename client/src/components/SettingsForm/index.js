import React from 'react';

// import components here
import { withStyles } from '@material-ui/core/styles';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';

import styles from './styles';

// import css here
// import './SettingsForm.css';

class SettingsForm extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      email: '',
      name: '',
      phone: ''
    };
  }
  changeHandler = e => {
    return this.setState({ [e.target.name]: e.target.value });
  };
  newPasswordSubmit = e => {
    e.preventDefault();
    const { newPassword, confirmPassword } = this.state;
    // change alerts when we start adding redux and axios calls
    newPassword === confirmPassword && newPassword.length > 7
      ? alert('Password Changed')
      : alert('Passwords do not match or invalid');
  };
  render() {
    const { name, email, phone } = this.state;
    const { classes } = this.props;
    return (
      <form className="form-column" onSubmit={this.newPasswordSubmit}>
        <TextField
          InputProps={{
            inputProps: {
              className: classes.textField
            }
          }}
          InputLabelProps={{
            className: classes.label
          }}
          style={{ fontSize: '2rem' }}
          label="Name"
          name="name"
          onChange={this.changeHandler}
          value={name}
          placeholder="john doe"
        />
        <TextField
          InputProps={{
            inputProps: {
              className: classes.textField
            }
          }}
          InputLabelProps={{
            className: classes.label
          }}
          style={{ fontSize: '2rem' }}
          label="Email"
          name="email"
          onChange={this.changeHandler}
          value={email}
          placeholder="testing@gmail.com"
        />
        <TextField
          InputProps={{
            inputProps: {
              className: classes.textField
            }
          }}
          InputLabelProps={{
            className: classes.label
          }}
          style={{ fontSize: '2rem' }}
          label="Phone"
          name="phone"
          onChange={this.changeHandler}
          value={phone}
          placeholder="6789995566"
          pattern="[0-9]{10}"
          maxLength="10"
        />
        <Button variant="contained" color="secondary">
          Save
        </Button>
      </form>
    );
  }
}

export default withStyles(styles)(SettingsForm);
