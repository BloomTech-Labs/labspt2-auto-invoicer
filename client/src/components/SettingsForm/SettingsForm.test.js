import React from "react";
import { render } from "react-testing-library";

import SettingsForm from "../SettingsForm";

describe("<SettingsForm />", () => {
  it("renders Settings Form", () => {
    render(<SettingsForm />);
  });
});
