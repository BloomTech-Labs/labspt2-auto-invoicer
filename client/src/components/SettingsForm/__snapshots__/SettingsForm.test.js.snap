// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SettingsForm /> renders Settings Form 1`] = `
<DocumentFragment>
  <form
    class="settings_form"
  >
    <fieldset
      class="settings_info"
    >
      <input
        name="email"
        placeholder="Email Address"
        type="email"
        value=""
      />
      <input
        name="oldPassword"
        placeholder="Old Password"
        type="password"
        value=""
      />
      <input
        name="newPassword"
        placeholder="New Password"
        type="password"
        value=""
      />
      <input
        name="confirmPassword"
        placeholder="Confirm Password"
        type="password"
        value=""
      />
      <button
        class="button"
        type="submit"
      >
        save
      </button>
      <p>
        Password must contain at least 8 charachters
      </p>
    </fieldset>
  </form>
</DocumentFragment>
`;
